pkg/ebpf/tracer_windows.go

```diff
@@ -2,23 +2,149 @@
 
 package ebpf
 
+/*
+typedef struct _stats
+{
+    long Read_calls;		//! number of read calls to the driver
+    long Read_bytes;
+    long Read_calls_outstanding;
+    long Read_calls_completed;
+    long Read_calls_cancelled;
+    long Read_packets_skipped;
+    long Write_calls;	//! number of write calls to the driver
+    long Write_bytes;
+    long Ioctl_calls;	//! number of ioctl calls to the driver
+} STATS;
+
+typedef struct driver_stats
+{
+    STATS		Total;		//! stats since the driver was started
+    STATS		Handle;		//! stats for the file handle in question
+} DRIVER_STATS;
+*/
+import "C"
 import (
+	"expvar"
+	"fmt"
+	"syscall"
+	"time"
+	"unsafe"
+
 	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	DRIVERFILE = `\\.\ddfilter`
+
+	// https://github.com/DataDog/datadog-windows-filter/blob/master/include/ddfilterapi.h
+	DDFILTER_IOCTL_GETSTATS               = 0x801
+	DDFILTER_IOCTL_SIMULATE_COMPLETE_READ = 0x802
+	DDFILTER_IOCTL_SET_FILTER             = 0x803
+
+	// https://docs.microsoft.com/en-us/windows-hardware/drivers/kernel/specifying-device-types
+	NETWORK_DEVICE_TYPE_CTL_CODE = 0x00000012
+)
+
+var (
+	kernel32        = syscall.MustLoadDLL("kernel32.dll")
+	CreateFile      = kernel32.MustFindProc("CreateFileW")
+	CloseHandle     = kernel32.MustFindProc("CloseHandle")
+	expvarEndpoints map[string]*expvar.Map
+	expvarTypes     = []string{"driver_total_stats", "driver_handle_stats"}
 )
 
+func init() {
+	expvarEndpoints = make(map[string]*expvar.Map, len(expvarTypes))
+	for _, name := range expvarTypes {
+		expvarEndpoints[name] = expvar.NewMap(name)
+	}
+}
+
 // Tracer struct for tracking network state and connections
 type Tracer struct {
-	config *Config
+	config       *Config
+	driverHandle syscall.Handle
 }
 
 // NewTracer returns an initialized tracer struct
 func NewTracer(config *Config) (*Tracer, error) {
-	return &Tracer{}, nil
+	handle, err := openDriverFile(DRIVERFILE)
+	if err != nil {
+		return nil, fmt.Errorf("%s : %s", "Could not create driver handle", err)
+	}
+	tr := &Tracer{
+		driverHandle: handle,
```

Comment:
Just to verify: do we need a separate handle for getting stats and getting packets? If so, let's call this stats handle.

Reply:
I'm not actually sure if it's necessary. I'm going to pull in @derekwbrown here since I think he would have a better idea on whether or not a handle separate from the one that will pull packets should be used to pull stats.

Reply:
it's a matter of implementation which is easier; same or different handle will work fine.